doc(virtualization-proxmox-ve-restapi): Refresh documentation page CTOR-1336 (#4036)
centreon/centreon-documentation: pp/integrations/plugin-packs/procedures/virtualization-proxmox-ve-restapi.md

import Tabs from '@theme/Tabs';
import TabItem from '@theme/TabItem';

## Connector dependencies

The following monitoring connectors will be installed when you install the **Proxmox VE** connector through the

> **Configuration > Monitoring Connector Manager** menu:

Reviewer: ```suggestion
**Configuration > Connectors > Monitoring Connectors** menu:
```

* [Base Pack](./base-generic.md)

## Pack assets

### Templates

The Monitoring Connector **Proxmox VE** brings a host template:

* **Virt-Proxmox-Ve-Restapi-custom**

The connector brings the following service templates (sorted by the host template they are attached to):

<Tabs groupId="sync">
<TabItem value="Virt-Proxmox-Ve-Restapi-custom" label="Virt-Proxmox-Ve-Restapi-custom">

| Service Alias | Service Template                             | Service Description  | Discovery |
|:--------------|:---------------------------------------------|:---------------------|:---------:|
| Node-Usage    | Virt-Proxmox-Ve-Node-Usage-Restapi-custom    | Check node usage     |     X     |
| Storage-Usage | Virt-Proxmox-Ve-Storage-Usage-Restapi-custom | Check stockage usage |     X     |

> | Vm-Usage      | Virt-Proxmox-Ve-Vm-Usage-Restapi-custom      | Check vms usage      |     X     |

Reviewer: ```suggestion
| Vm-Usage      | Virt-Proxmox-Ve-Vm-Usage-Restapi-custom      | Check VM usage      |     X     |
```

> The services listed above are created automatically when the **Virt-Proxmox-Ve-Restapi-custom** host template is used.

> If **Discovery** is checked, it means a service discovery rule exists for this service template.

</TabItem>
</Tabs>

### Discovery rules

#### Host discovery

| Rule name       | Description                      |
|:----------------|:---------------------------------|

> | Proxmox VM      | Discover Proxmox Virtual Machine |

Reviewer: ```suggestion
| Proxmox VM      | Discover Proxmox virtual machines |
```

> To monitor, a user with read privileges is required: `VM.Monitor`, `VM.Audit`, `Datastore.Audit`, `Sys.Audit`, `Sys.Syslog`.

Reviewer: ```suggestion
To monitor, a user with the following "read" privileges is required: `VM.Monitor`, `VM.Audit`, `Datastore.Audit`, `Sys.Audit`, `Sys.Syslog`.
```

To monitor, a user with read privileges is required: `VM.Monitor`, `VM.Audit`, `Datastore.Audit`, `Sys.Audit`, `Sys.Syslog`.

> Please refer to their (official documentation)[https://pve.proxmox.com/wiki/Proxmox_VE_API].

Author: ```suggestion
Please refer to their [official documentation](https://pve.proxmox.com/wiki/Proxmox_VE_API).
```

## Installing the monitoring connector

### Pack

> 

Reviewer: ```suggestion

The installation procedures for monitoring connectors are slightly different depending on [whether your license is offline or online](../getting-started/how-to-guides/connectors-licenses.md).

```

1. If the platform uses an *online* license, you can skip the package installation
instruction below as it is not required to have the connector displayed within the

> **Configuration > Monitoring Connector Manager** menu.

Reviewer: ```suggestion
**Configuration > Connectors > Monitoring Connectors** menu.
```

If the platform uses an *offline* license, install the package on the **central server**
with the command corresponding to the operating system's package manager:

<Tabs groupId="sync">
<TabItem value="Alma / RHEL / Oracle Linux 8" label="Alma / RHEL / Oracle Linux 8">

```bash
dnf install centreon-pack-virtualization-proxmox-ve-restapi
```

</TabItem>
<TabItem value="Alma / RHEL / Oracle Linux 9" label="Alma / RHEL / Oracle Linux 9">

```bash
dnf install centreon-pack-virtualization-proxmox-ve-restapi
```

</TabItem>
<TabItem value="Debian 11 & 12" label="Debian 11 & 12">

```bash
apt install centreon-pack-virtualization-proxmox-ve-restapi
```

</TabItem>
<TabItem value="CentOS 7" label="CentOS 7">

```bash
yum install centreon-pack-virtualization-proxmox-ve-restapi
```

</TabItem>
</Tabs>

2. Whatever the license type (*online* or *offline*), install the **Proxmox VE** connector through

> the **Configuration > Monitoring Connector Manager** menu.

Reviewer: ```suggestion
the **Configuration > Connectors > Monitoring Connectors** menu.
```

```bash
apt install centreon-plugin-virtualization-proxmox-ve-restapi
```

</TabItem>
<TabItem value="CentOS 7" label="CentOS 7">

```bash
yum install centreon-plugin-Virtualization-Proxmox-Ve-Restapi
```

</TabItem>
</Tabs>

## Using the monitoring connector

### Using a host template provided by the connector

1. Log into Centreon and add a new host through **Configuration > Hosts**.
2. Fill in the **Name**, **Alias** & **IP Address/DNS** fields according to your resource's settings.
3. Apply the **Virt-Proxmox-Ve-Restapi-custom** template to the host. A list of macros appears. Macros allow you to define how the connector will connect to the resource, and to customize the connector's behavior.
4. Fill in the macros you want. Some macros are mandatory.

| Macro                  | Description                                                                                                                              | Default value     | Mandatory |
|:-----------------------|:-----------------------------------------------------------------------------------------------------------------------------------------|:------------------|:---------:|

> | PROXMOXAPIUSERNAME     | Set Proxmox VE Username API user need to have this privileges 'VM.Monitor, VM.Audit, Datastore.Audit, Sys.Audit, Sys.Syslog'             |                   |     X     |

Reviewer: ```suggestion
| PROXMOXAPIUSERNAME     | Set Proxmox VE Username. The API user needs to have the following privileges: 'VM.Monitor, VM.Audit, Datastore.Audit, Sys.Audit, Sys.Syslog'             |                   |     X     |
```

> OK: Storage 'storage/nuc/local' state: available, space total: 217.61 GB used: 145.86 GB (67.03%) free: 71.76 GB (32.97%) | 'storage/nuc/local#storage.space.usage.bytes'=156610641920B;;;0;233658822656

Reviewer: thresholds?

> OK: Storage 'storage/nuc/local' state: available, space total: 217.61 GB used: 145.86 GB (67.03%) free: 71.76 GB (32.97%) | 'storage/nuc/local#storage.space.usage.bytes'=156610641920B;;;0;233658822656

Author: ```suggestion
OK: Storage 'storage/nuc/local' state: available, space total: 217.61 GB used: 145.86 GB (67.03%) free: 71.76 GB (32.97%) | 'storage/nuc/local#storage.space.usage.bytes'=156610641920B;0:186927058124;0:210292940390;0;233658822656
```
